libbeat/outputs/elasticsearch/client.go

```diff
@@ -53,6 +55,14 @@ type ClientSettings struct {
 	Pipeline           *outil.Selector
 	Timeout            time.Duration
 	CompressionLevel   int
+	Stats              *ClientStats
+}
+
+type ClientStats struct {
```

Comment:
I'd prefer (also for metrics), using structs/interfaces to be passed to output clients, instead of globals. I introduced the `ClientStats` for the ES client, as I wanted to ensure, these stats not being accidentally used by the ES monitoring reporter.
By passing stats by struct, we can register/unregister metrics in the future, when reporting stats + reloading outputs.